### ISSUE TITLE
CSV file upload doesn't correctly handle "UTF-8 with BOM" CSVs

### ISSUE THREAD

**Issue by MeganerdDev** [nautobot/nautobot#5812]
### Environment
* Nautobot version (Docker tag too if applicable): v2.2.25
* Python version: 3.11
* Database platform, version: PG15
* Middleware(s): N/A

### Steps to Reproduce
1. Affecting [ipam/prefixes](https://demo.nautobot.com/ipam/prefixes/) CSV Import feature
2. Upload a bulk Prefix CSV file to `Prefix` that contains the columns marked as required: prefix, status
![image](https://github.com/nautobot/nautobot/assets/114895043/43bd5246-e93e-4671-b4a7-5e4b6a949638)
4. The job will get an error: `Row <id>: prefix : this field is required.`
![image](https://github.com/nautobot/nautobot/assets/114895043/4069cfd6-03e0-4410-a72e-5357a6db17e8)

### Expected Behavior
Upload is parsed without error, job completes

### Observed Behavior
job error: `Row <id>: prefix : this field is required.`

### Mitigation
The unspecified column `display` is included in the CSV export (including the `prefix` column separately)
For the CSV Upload to complete I mirrored this, and include both a `display` and `prefix` colum in my CSV upload

### Headers from Prefix CSV Export
`display,id,object_type,natural_slug,prefix,type,network,broadcast,prefix_length,ip_version,date_allocated,description,status__name,role__name,parent__namespace__name,parent__network,parent__prefix_length,namespace__name,tenant__name,vlan__pk,rir__name,created,last_updated,tags`

**Comment by glennmatthews** [comment]
I'm not sure I follow - are you saying that if you omit the `display` column, you get an error about the `prefix` column? Can you provide a CSV file that demonstrates the issue?

**Comment by MeganerdDev** [comment]
| are you saying that if you omit the display column, you get an error about the prefix column
Correct, this is what I was able to reproduce this locally. I just tried again now in the demo v2.2.5 instance an did not see the same behavior. I will perform more testing and update/confirm this.

**Comment by gsnider2195** [comment]
This is working for me today on demo.nautobot.com

![image](https://github.com/nautobot/nautobot/assets/75227981/3425c10a-9cc1-4e50-bee2-86f0a6b13396)

![image](https://github.com/nautobot/nautobot/assets/75227981/1bc5ae0b-eb4d-4e08-9304-c8008330fc7b)

@MeganerdDev if you can reproduce on the demo instance let us know

**Comment by cdwchriburg** [comment]
@gsnider2195 The test you did is not completely valid.  It appears you must upload the CSV file, not provide CSV text input to hit this issue.  There appears to be some behavior different there.

I am facing same issue on 2.2.5.  Reproducible on demo.nautobot.com running 2.2.8.

Create a CSV File with this contents and attempt to import from CSV file upload:
```
prefix,status
192.168.1.1/32,Active
```

This will fail.  The same CSV text put into CSV Text Input field works.


Workaround for CSV file import to work is to create a CSV file with this data:
```
display,prefix,status
192.168.1.1/32,192.168.1.1/32,Active
```

Note that the column order matters as well.  If `display` column is present, but it is not the first column, then CSV File Upload import will also fail.

This broke sometime recently - somewhere between 2.1.7 and 2.2.5.  Recently have done a ton of CSV imports where this issue was not faced on the earlier 2.1.7 code.

**Comment by glennmatthews** [comment]
That's really weird, as there's virtually no difference between the code paths for text field versus file upload:

```py
        if csv_file:
            csv_bytes = csv_file
        else:
            csv_bytes = BytesIO(csv_data.encode("utf-8"))
```

I'll see if I can reproduce this locally and throw in a few more debugs to troubleshoot.

**Comment by glennmatthews** [comment]
Worked here on 2.2.7 with file upload:

![image](https://github.com/user-attachments/assets/3eaf9ac1-ec3b-48ee-a1e4-06f642980da1)

![image](https://github.com/user-attachments/assets/152a7ce5-4070-4c83-93d6-84d34d688902)

Also worked today on demo.nautobot.com:

![image](https://github.com/user-attachments/assets/21a10906-5e99-48e7-923f-476bb05f2256)

![image](https://github.com/user-attachments/assets/ec1e515e-e628-4088-a8e8-1291fe9b96c6)

If you're able to reproduce this again, could you capture the `Job keyword arguments` from the "Additional Data" tab of the failed JobResult?

**Comment by glennmatthews** [comment]
Aha! Thank you for chasing that down. It looks like this might be addressable by opening the file with encoding `utf-8-sig` instead of `utf-8`.

- Django 3 and 4 don't appear to offer an `encoding` option when opening a `FileField`/`FieldFile` (may be addressed in 5.0 by https://github.com/django/django/pull/17173)
- It may be possible/desirable for the Job to use a `codecs.EncodedFile` (a la https://stackoverflow.com/questions/4729832/processing-a-django-uploadedfile-as-utf-8-with-universal-newlines) to guarantee that the file encoding is correct
